src/deadline_worker_agent/sessions/actions/scripts/attachment_upload.py

```diff
@@ -48,26 +48,32 @@ def merge(manifest_paths_by_root: dict[str, list[str]]) -> dict[str, str]:
         if len(paths) == 1:
             merged_manifests[root] = paths[0]
         else:
+            file_name_list = [os.path.basename(path) for path in paths]
+            # generating the name while keepign the original hash
+            name = "_".join(set("_".join(file_name_list).split("_")))
```

Comment:
```suggestion
            name = "_".join(set((file_name_list)))
```

Any reason we're joining by `_` and spitting by it again?

Reply:
Take `file_name_list = ['40ae15f70df956ca6fb2f5f1cdc6cdfb_job', '40ae15f70df956ca6fb2f5f1cdc6cdfb_step']` as an example, implementation in the PR produces `job_step_40ae15f70df956ca6fb2f5f1cdc6cdfb`, while the suggested one produces `40ae15f70df956ca6fb2f5f1cdc6cdfb_step_40ae15f70df956ca6fb2f5f1cdc6cdfb_job`. The current one seems cleaner, but I don't mind too much as it's short lived under session directory.

Reply:
Ah okay, I was just confused at the syntax. Looked weird to join by `_` then split by `_`, seemed like a no-op. It wasn't clear that we wanted to split the filenames with `_` in them as well. We could do something like this to be more clear, but not necessary:

```py
name = "_".join(set([part for file_name in file_name_list for part in file_name.split("_")]))
```